nil pointer dereference in "TLS Validation in Destination Rule" destinationrule Auditor

- Snowcat version: 0.1.0 Linux AMD64
- Istio version: 1.11.2

[the deref looks to be here](https://github.com/praetorian-inc/snowcat/blob/main/auditors/destinationrule/cacerts.go#L47)

```
# ./snowcat -l debug
WARN[0000] configuration file not found
INFO[2021-10-14T22:58:07Z] successfully loaded config file               configFile=snowcat.yml
INFO[2021-10-14T22:58:07Z] running discovery strategy                    runner=Kubelet strategy=default-gateway
INFO[2021-10-14T22:58:07Z] attempting to locate default gateway
INFO[2021-10-14T22:58:07Z] discovered default gateway                    gateway=169.254.1.1
INFO[2021-10-14T22:58:07Z] scanning for additional potential gateways using HTTP scanner
DEBU[2021-10-14T22:58:07Z] discovered kubelet api                        addr="169.254.123.1:10255"
DEBU[2021-10-14T22:58:07Z] resulting kubelet apis                        kubeletAddresses="[169.254.123.1:10255]"
INFO[2021-10-14T22:58:07Z] running discovery strategy                    runner=Namespace strategy=envoy
DEBU[2021-10-14T22:58:07Z] sending HTTP request to envoy                 method=GET url="http://localhost:15000/config_dump"
INFO[2021-10-14T22:58:07Z] running discovery strategy                    runner="Istio Control Plane" strategy=kubelet
DEBU[2021-10-14T22:58:07Z] connecting to xds                             addr="10.0.17.11:15010"
DEBU[2021-10-14T22:58:07Z] connecting to xds                             addr="10.0.17.11:15010"
DEBU[2021-10-14T22:58:07Z] validating debug API with HTTP request        method=HEAD url="http://10.0.17.11:8080/debug/configz"
DEBU[2021-10-14T22:58:07Z] connecting to xds                             addr="10.0.17.11:15010"
INFO[2021-10-14T22:58:08Z] running auditor                               auditor="Deny with Positive Match"
INFO[2021-10-14T22:58:08Z] running auditor                               auditor="TLS Validation in Destination Rule"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x138e41b]

goroutine 1 [running]:
github.com/praetorian-inc/snowcat/auditors/destinationrule.(*auditor).Audit(0x2824df0, 0xc001b876c0, 0x6, 0xc0008811e7, 0xc, 0xc001b86730, 0x10, 0x0, 0x0, 0xc000ab4990, ...)
        /home/runner/work/snowcat/snowcat/auditors/destinationrule/cacerts.go:47 +0x57b
github.com/praetorian-inc/snowcat/pkg/cli.RunSnowcat(0xc0002b6660, 0x0, 0x2)
        /home/runner/work/snowcat/snowcat/pkg/cli/root.go:250 +0x5d5
github.com/praetorian-inc/snowcat/pkg/cli.glob..func2(0x27dc880, 0xc0002b6660, 0x0, 0x2)
        /home/runner/work/snowcat/snowcat/pkg/cli/root.go:85 +0x3f
github.com/spf13/cobra.(*Command).execute(0x27dc880, 0xc00003a190, 0x2, 0x2, 0x27dc880, 0xc00003a190)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:860 +0x2c2
github.com/spf13/cobra.(*Command).ExecuteC(0x27dc880, 0x7feee0d4f748, 0x0, 0x40e0f8)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:974 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:902
github.com/praetorian-inc/snowcat/pkg/cli.Execute()
        /home/runner/work/snowcat/snowcat/pkg/cli/root.go:171 +0x45
main.main()
        /home/runner/work/snowcat/snowcat/cmd/snowcat/main.go:24 +0x25
 ```

--- a/_fixtures/destinationrule-empty-tls.yaml
+++ b/_fixtures/destinationrule-empty-tls.yaml
@@ -1,0 +1,8 @@
+apiVersion: networking.istio.io/v1alpha3
+kind: DestinationRule
+metadata:
+  name: empty-tls-policy
+  namespace: default
+spec:
+  host: example.com
+  trafficPolicy: {}

--- a/_fixtures/destinationrule-empty-traffic-policy.yaml
+++ b/_fixtures/destinationrule-empty-traffic-policy.yaml
@@ -1,0 +1,7 @@
+apiVersion: networking.istio.io/v1alpha3
+kind: DestinationRule
+metadata:
+  name: empty-traffic-policy
+  namespace: default
+spec:
+  host: example.com

--- a/auditors/destinationrule/cacerts.go
+++ b/auditors/destinationrule/cacerts.go
@@ -44,6 +44,9 @@ func (a *auditor) Audit(_ types.Discovery, resources types.Resources) ([]types.A
 	var results []types.AuditResult
 
 	for _, rule := range resources.DestinationRules {
+		if rule.Spec.TrafficPolicy == nil {
+			continue
+		}
 		if !isClientTLSSettingSafe(rule.Spec.TrafficPolicy.Tls) {
 			results = append(results, types.AuditResult{
 				Name:        a.Name(),

